### ISSUE TITLE
Unity's own tests won't compile if variadic macros aren't supported

### ISSUE THREAD

**Issue by detly** [ThrowTheSwitch/Unity#418]
If you try to compile and run Unity's own tests on a compiler that doesn't support variadic macros, they won't compile eg. on TI's CGT clxx:

```
build/testunityRunner.c: In function ‘main’:
build/testunityRunner.c:464:53: error: macro "RUN_TEST" passed 2 arguments, but takes just 1
     RUN_TEST(testUnitySizeInitializationReminder, 86);
                                                     ^
```

(Note that this compiler DOES support variadic macros but Unity doesn't know that, so attempts to use non-variadic forms. It's just for illustration. You should be able to reproduce this with any compiler in strict ANSI mode.)

**Comment by detly** [comment]
This is still an issue. It bites me every time I update to see if Unity has upstream support for TI.

**Comment by mvandervoord** [comment]
Which target are you specifying when you try to run the self tests? 

There isn't one in the list for TI c55x, but if we add one, clearly we wouldn't want to run the tests for parameterized tests because the compiler wouldn't support it. 

Would this solve your problem or am I missing the point?

**Comment by detly** [comment]
> Which target are you specifying when you try to run the self tests? 

I'm not sure what you mean by this.

The TI compiler can be set to support variadic macros or not to.

If I set it not to (strict ANSI mode), the tests straight up don't compile because the support for non-variadic macro operation seems to be broken (I think... it's complicated to set this up for compilation by CCS).

If I set it to support variadic macros (allow GCC extensions + relaxed parsing), they still don't compile because the `__STDC_VERSION__ >= 199901L` check doesn't perfectly map to "supports variadic macros" - the TI compiler doesn't set it.

So one bug is: if Unity intends to support running its own tests on targets without support for variadic macros, it doesn't* which is a bug. (*again, I'm not 100% sure about this, so feel free to verify)

Another bug is: when compiling Unity's own tests on a compiler that supports variadic macros but doesn't set `__STDC_VERSION__ >= 199901L`, variadic macros are not used, which should probably also be a bug. It's fixed by [this](https://github.com/detly/Unity/commit/138dc8175d2d75a1e5c5f408bcdf1e8b277742ad) and [this](https://github.com/detly/Unity/commit/132261d6146a39be9a3fcac48c84e6fb3ea55d27) if you're interested.
